inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/config/model/instrumentation/scope/TypeScope.java

```diff
@@ -28,8 +28,8 @@
     private NameMatcherSettings superclass;
 
     /**
-     * Matcher which have to match the class name.
+     * Matcher which have to match the type's name.
      */
-    private List<NameMatcherSettings> classes;
+    private List<NameMatcherSettings> types;

```

Comment:
Shouldn't it be "type" instead of "types"? this is why you didn't like "classes", right?

Reply:
Also: why is this a list while superclass is not a list? "Types" makes it sound like you can match for a disjunction of matchers while the NameMatchers are actually combined using conjunction.

Reply:
I think it does not makes sense to be able to define more than one superclass because all of them would have to match, so you can define the "lowest" one in that chain to achieve the same result.

Renamed the field into "type"

Reply:
What I mean: you could have something similar as for type, e.g. "The class must have a superclass from package xyz ending with ABC". Matching the superclass is conceptually almost the same as matching the type, therefore it makes sense to have a list of name matchers in either both or none.

Reply:
As discussed, we'll cahnge it to a single matcher because using regex (yes it is more expensive) the same result can be achieved and the config is more consistent